components/containers/MessagesContainer/MessagesContainer.js

```diff
@@ -0,0 +1,144 @@
+import * as Styles from './MessagesContainer.styles'
+import Messages from '../../Messages/Messages'
+import ViewMessageDialog from '../../dialogs/ViewMessageDialog/ViewMessageDialog'
+import ComposeMessageDialog from '../../dialogs/ComposeMessageDialog/ComposeMessageDialog'
+import SnackBar from '../../SnackBar/SnackBar'
+import {
+	viewMessageDialogOpen,
+	viewMessageDialogClosed,
+} from '../../../redux/viewMessageDialog'
+import {
+	composeMessageDialogOpen,
+	composeMessageDialogClosed,
+	messageFormChange,
+	messageFormSubmit,
+	sendRequestSuccess,
+	sendRequestFail,
+} from '../../../redux/composeMessageDialog'
+import { receivedMessageDeleted } from '../../../redux/reducers'
+import { alert, onAlertClose } from '../../../redux/snackbar'
+import { useDispatch, useSelector } from 'react-redux'
+import axios from 'axios'
+import { Box } from '@material-ui/core'
+
+const sendMessageUrl = '/api/messages/sendMessage/'
+const deleteMessageUrl = '/api/messages/removeMessage/'
+
+export default function MessagesContainer(props) {
+	const dispatch = useDispatch()
+	const viewMessageDialogState = useSelector((state) => state.viewMessageDialog)
```

Comment:
I would recommend looking into redux `connect` instead of `useSelector` and make your component a HOC. It's more efficient as useSelector tends to re-render a lot, but thats more of a scaling thing to be aware of. It also clears up all this logic in your render.